### PR TITLE
SONAR-7219 Remove sonar.security.updateUserAttributes property

### PR DESCRIPTION
**Pull request by julienlancelot**
SONAR-7219 Remove sonar.security.updateUserAttributes property

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java [hunk 1]
```diff
@@ -240,11 +240,6 @@
   String CORE_AUTHENTICATOR_IGNORE_STARTUP_FAILURE = "sonar.authenticator.ignoreStartupFailure";
   String CORE_AUTHENTICATOR_CREATE_USERS = "sonar.authenticator.createUsers";
 
-  /**
-   * @since 3.6
-   */
-  String CORE_AUTHENTICATOR_UPDATE_USER_ATTRIBUTES = "sonar.security.updateUserAttributes";
```
Comment (simonbrandhof) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: I suggest to keep the constant deprecated and to document the fact that feature is not supported anymore.

**Comment by simonbrandhof** [comment]
Minor feedback, else LGTM

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java [hunk 2]
```diff
@@ -242,7 +242,9 @@
 
   /**
    * @since 3.6
+   * @deprecated since 5.4. This feature is not supported anymore. See http://jira.sonarsource.com/browse/SONAR-7219
    */
+  @Deprecated
   String CORE_AUTHENTICATOR_UPDATE_USER_ATTRIBUTES = "sonar.security.updateUserAttributes";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Pull request merged**